datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/restoreindices/SendMAEStep.java

```diff
@@ -111,42 +122,185 @@ public Function<UpgradeContext, UpgradeStepResult> executable() {
+        context.report().addLine(String.format("Found aspectName arg as %s", aspectName));
+      } else {
+        aspectName = Optional.empty();
+        context.report().addLine("No aspectName arg present");
+      }
+      Optional<String> urn;
+      if (containsKey(context.parsedArgs(), RestoreIndices.URN_ARG_NAME)) {
+        urn = context.parsedArgs().get(RestoreIndices.URN_ARG_NAME);
+        context.report().addLine(String.format("Found urn arg as %s", urn));
+      } else {
+        urn = Optional.empty();
+        context.report().addLine("No urn arg present");
+      }
+
+      ThreadPoolExecutor executor = (ThreadPoolExecutor) Executors.newFixedThreadPool(numThreads);
+
+      context.report().addLine(
+              String.format("Sending MAE from local DB with %s batch size, %s threads, %s batchDelayMs",
+                      batchSize, numThreads, batchDelayMs));
+      ExpressionList<EbeanAspectV2> countExp =
+          _server.find(EbeanAspectV2.class)
+            .where()
+            .eq(EbeanAspectV2.VERSION_COLUMN, ASPECT_LATEST_VERSION);
+      if (aspectName.isPresent()) {
+        countExp = countExp.eq(EbeanAspectV2.ASPECT_COLUMN, aspectName.get());
+      }
+      if (urn.isPresent()) {
+        countExp = countExp.eq(EbeanAspectV2.URN_COLUMN, urn.get());
+      }
+      final int rowCount = countExp.findCount();
+      context.report().addLine(String.format("Found %s latest aspects in aspects table", rowCount));
+
+      int totalRowsMigrated = 0;
+      int start = 0;
+      int ignored = 0;
+
+      List<Future<KafkaJobResult>> futures = new ArrayList<>();
+      while (start < rowCount) {
+        while (futures.size() < numThreads) {
+          futures.add(executor.submit(new KafkaJob(context, start, batchSize, batchDelayMs, aspectName, urn)));
+          start = start + batchSize;
+        }
+        KafkaJobResult tmpResult = iterateFutures(futures);
+        if (tmpResult != null) {
+          totalRowsMigrated += tmpResult.rowsMigrated;
+          ignored += tmpResult.ignored;
+          reportStats(context, totalRowsMigrated, ignored, rowCount, startTime);
+        }
+      }
+      while (futures.size() > 0) {
+        KafkaJobResult tmpResult = iterateFutures(futures);
+        if (tmpResult != null) {
+          totalRowsMigrated += tmpResult.rowsMigrated;
+          ignored += tmpResult.ignored;
+          reportStats(context, totalRowsMigrated, ignored, rowCount, startTime);
+        }
+      }
+      executor.shutdown();
       if (totalRowsMigrated != rowCount) {
-        context.report().addLine(String.format("Failed to send MAEs for %d rows...", rowCount - totalRowsMigrated));
+        float percentFailed = 0.0f;
+        if (rowCount > 0) {
+          percentFailed = (float) (rowCount - totalRowsMigrated) * 100 / rowCount;
+        }
+        context.report().addLine(String.format(
+                "Failed to send MAEs for %d rows (%.2f%%).", rowCount - totalRowsMigrated, percentFailed));
```

Comment:
I think the percent should be clarified here, generally I think people would expect to see the percent completed not percent failed so something like

Failed to send MAEs for %d rows, (%.2f%%) of total failed